adal/src/main/java/com/microsoft/aad/adal/HashMapExtensions.java

```diff
@@ -110,4 +113,36 @@ private HashMapExtensions() {
         return response;
     }
 
+    static HashMap<String, String> jsonStringAsMap(String bodyStr) throws JSONException {
+        final HashMap<String, String> responseItems = new HashMap<>();
```

Comment:
Let's also consider changing the return type to `Map` -- I know the class is called _HashMapExtensions_, but we should still try to code to an interface where possible :-)